app/services/ems_physical_infra_dashboard_service.rb

```diff
@@ -101,39 +101,45 @@ def recent_servers
     }
   end
 
-  def notification_data(attrs)
-    valid = 0
-    warning = 0
-    critical = 0
-    attrs.each do |attr|
-      next unless attr.respond_to?(:health_state)
-      case attr.health_state&.downcase
-      when 'critical'
-        critical += 1
-      when 'warning'
-        warning += 1
-      when 'valid'
-        valid += 1
-      end
-    end
-    if critical.positive?
-      icon_class = 'pficon pficon-error-circle-o'
-      count = critical
-    elsif warning.positive?
-      icon_class = 'pficon pficon-warning-triangle-o'
-      count = warning
-    elsif valid.positive?
-      icon_class = 'pficon pficon-ok'
-      count = valid
-    else
-      icon_class = 'pficon pficon-error-circle-o'
+  def notification_data(components, component_type)
+    if components&.first.respond_to?(:health_state)
       count = 0
+      health_states = components.group('lower(health_state)').count
+      health_states.default = 0
+      critical_count = health_states["critical"]
+      warning_count = health_states["warning"]
+
+      if critical_count.positive?
+        icon_class = 'pficon pficon-error-circle-o'
+        count = critical_count
+        health_state = 'critical'
+      elsif warning_count.positive?
+        icon_class = 'pficon pficon-warning-triangle-o'
+        count = warning_count
+        health_state = 'warning'
+      else
+        health_state = 'valid'
+        icon_class = 'pficon pficon-ok'
+      end
+      tooltip_count = count.positive? ? count : components.count
+      {
+        :count     => count,
+        :iconClass => icon_class,
+        :tooltip   => format_tooltip(health_state, component_type.downcase, tooltip_count),
+      }
     end
+  end
 
-    {
-      :iconClass => icon_class,
-      :count     => count,
+  def format_tooltip(health_state, component_type, count)
+    singular_structure = "%{count} %{component_type} is in %{health_state} state."
+    plural_structure = "%{count} %{component_type} are in %{health_state} state."
```

Comment:
@mzazrivec do we have something for this is/are thing?

Reply:
This is fine. Though the strings need to be inside `n_()`. `n_(var1, var2, count)` won't work with the string extraction process.

Reply:
@mzazrivec just a question, why `n_(var1, var2, count)` won't work? There must be only one string per `n_( )` ?

Reply:
The extraction logic cannot parse the string out if it's in a variable (or any expression).

Reply:
@mzazrivec thanks. Is it fine now?

Reply:
Yes.

Reply:
It's better than fine, it's gaprindashvili! :laughing:

Reply:
Keep hammering that home.

Reply:
@mzazrivec @skateman Could someone merge or approve this PR ?